staging/src/k8s.io/apimachinery/pkg/api/validation/generic_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import "testing"
+
+func TestMaskTrailingDash(t *testing.T) {
+	testCases := []struct {
+		beforeMasking        string
+		expectedAfterMasking string
+		description          string
+	}{
+		{
+			beforeMasking:        "",
+			expectedAfterMasking: "",
+			description:          "empty string",
+		},
+		{
+			beforeMasking:        "-",
+			expectedAfterMasking: "-",
+			description:          "only a single dash",
+		},
+		{
+			beforeMasking:        "-foo",
+			expectedAfterMasking: "-foo",
+			description:          "has leading dash",
+		},
+		{
+			beforeMasking:        "-foo-",
+			expectedAfterMasking: "-foa",
```

Comment:
Why -foa instead of -foo-a

Reply:
mask trailing dash should not make the generate name longer or it could fail length validation